packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -892,6 +895,63 @@ class DropdownButton<T> extends StatefulWidget {
        assert(isExpanded != null),
        assert(autofocus != null),
        assert(itemHeight == null || itemHeight >=  kMinInteractiveDimension),
+       _inputDecoration = null,
+       _isEmpty = false,
+       _isFocused = false,
+       super(key: key);
+
+  DropdownButton._formField({
+    Key? key,
+    required this.items,
+    this.selectedItemBuilder,
+    this.value,
+    this.hint,
+    this.disabledHint,
+    required this.onChanged,
+    this.onTap,
+    this.elevation = 8,
+    this.style,
+    this.underline,
+    this.icon,
+    this.iconDisabledColor,
+    this.iconEnabledColor,
+    this.iconSize = 24.0,
+    this.isDense = false,
+    this.isExpanded = false,
+    this.itemHeight = kMinInteractiveDimension,
+    this.focusColor,
+    this.focusNode,
+    this.autofocus = false,
+    this.dropdownColor,
+    this.menuMaxHeight,
+    this.enableFeedback,
+    this.alignment = AlignmentDirectional.centerStart,
+    this.borderRadius,
+    required InputDecoration? inputDecoration,
+    required bool isEmpty,
+    required bool isFocused,
+    // When adding new arguments, consider adding similar arguments to
+    // DropdownButtonFormField.
```

Comment:
Nit. This seems unnecessary now (same in the public constructor).

Reply:
I am removing it for the private constructor which I added to this PR
For the public constructor, I think it is nice to have more documentation than less, in case somebody needs to know why they need to add those arguments, why it is "forced" by the analyzer.